td-shim-tools/src/bin/td-payload-reference-calculator/main.rs

```diff
@@ -132,6 +117,41 @@ fn get_image_regions(buf: &[u8]) -> (usize, Vec<usize>, Vec<usize>) {
     let size_of_optional_header = ((buf[coff_file_header_offset as usize + 17] as u16) << 8)
         | buf[coff_file_header_offset as usize + 16] as u16;
 
+    let optional_header_addr: usize = (coff_file_header_offset + size_of_coff_file_header) as usize;
+
+    // Refer to https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#optional-header-standard-fields-image-only
+    // Magic           Offset: 0       Size: 2  Note: 0x10b for PE32, ox20b for PE32+
+    //                 PE32/PE32+
+    // SizeOfHeaders   Offset: 60      Size: 4
+    // CheckSum        Offset: 64      Size: 4
+    // Cert table      Offset: 128/144     Size: 8
+
+    let magic = ((buf[optional_header_addr + 1] as u16) << 8) | buf[optional_header_addr] as u16;
+    let is_pe32 = is_pe32(magic).ok().unwrap();
+    let optional_size_of_headers_offset: usize = 0x003c;
+    let optional_checksum_offset: usize = 0x0040;
+    let optional_cert_table_offset = if is_pe32 { 0x0080 } else { 0x0090 };
```

Comment:
How do we support PE32 (IA32), not PE32+ (X64)?

Reply:
Only support PE32+, removed magic check